.github/workflows/check-release.yml

```diff
@@ -0,0 +1,170 @@
+
+      - name: Check if release has changed
+        id: check_release_changes
+        run: |
+          new_tag=${{ steps.release_outputs.outputs.tag_name }}
+          new_sha=${{ steps.release_outputs.outputs.sha }}
+
+          if [ -f release.json ]; then
+            current_tag=$(jq -r '.current_tag' release.json)
+            current_sha=$(jq -r '.current_sha' release.json)
+
+            if [[ "$current_tag" == "$new_tag" && "$current_sha" == "$new_sha" ]]; then
+              echo "No changes in release."
+              echo "release_changed=false" >> "$GITHUB_OUTPUT"
+            else
+              echo "Release has changed."
+              echo "release_changed=true" >> "$GITHUB_OUTPUT"
+              # Update release.json
+              jq -n --arg new_tag "$new_tag" --arg new_sha "$new_sha" \
+                '{current_tag: $new_tag, current_sha: $new_sha}' > release.json
+  
+              cat release.json
+            fi
+  
+          else
+            echo "Release is changed, file release.json does not exist."
+            echo "release_changed=true" >> "$GITHUB_OUTPUT"
+
+            # Create release.json
+            jq -n --arg new_tag "$new_tag" --arg new_sha "$new_sha" \
+              '{current_tag: $new_tag, current_sha: $new_sha}' > release.json
+
+            cat release.json
+          fi
+
+      # Retrieve the SHA of the latest release for the linux/amd64 architecture.
+      # To accomplish this, we fetch all versions and identify the 5th version, which corresponds to the latest release with that specific architecture.
+      # Reference: https://docs.github.com/en/rest/packages/packages?apiVersion=2022-11-28#list-package-versions-for-a-package-owned-by-an-organization
+      ## NOTE: The SHA obtained in this step differs from the previous one, as it represents the branch commit SHA used for change verification.
+
+      - name: Get Docker image digests
+        if: steps.check_release_changes.outputs.release_changed == 'true'
+        id: get_digests
+        run: |
+          digests=$(gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /orgs/actions/packages/container/actions-runner/versions)
+          latest_digest=$(echo "$digests" | jq -r '.[] | select(.metadata.container.tags | index("latest")) | .name')
+          echo $latest_digest
+          echo "arch_sha_name=$latest_digest" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_TOKEN: ${{ env.GITHUB_TOKEN }}
+
+      - name: Get changelog file
+        id: changelog
+        run: |
+          if [ ! -f CHANGELOG.md ]; then
+            echo "Changelog file does not exist. Creating one..."
+            touch CHANGELOG.md
+          fi
+
+      - name: Update Changelog
+        if: steps.check_release_changes.outputs.release_changed == 'true'
+        id: update_changelog
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
```

Comment:
it looks like this repo use semantic-release to update changelog using conventional commit mssages, may be we can reuse it someway here?

see https://github.com/semantic-release/release-notes-generator